File: ProblemSet_12xx/Problem_1200/solution.py
import math


class Solution:
    def minimumAbsDifference(arr: list[int]):
        arr.sort()
        result = math.inf
        resultArr = []
        for i in range(1, len(arr)):
            if (arr[i] - arr[i-1] <  result):
                result = arr[i] - arr[i-1]
                resultArr = [[arr[i-1], arr[i]]]
            elif (arr[i] - arr[i-1] == result):
                resultArr.append([arr[i-1], arr[i]])
        return resultArr

File: ProblemSet_12xx/Problem_1200/solution.cs
public class Solution {
    public IList<IList<int>> MinimumAbsDifference(int[] arr) {
        Array.Sort(arr);
        List<IList<int>> res = [];
        int min = int.MaxValue;
        for (int i = 0; i < arr.Length - 1; i++)
        {
            if (arr[i + 1] - arr[i] < min)
            {
                res = [[arr[i], arr[i + 1]]];
                min = arr[i + 1] - arr[i];
            }
            else if (arr[i + 1] - arr[i] == min)
            {
                res.Add([arr[i], arr[i + 1]]);
            }
        }
        return res;
    }
}